ReactAndroid/src/main/java/com/facebook/react/ReactDelegate.java

```diff
@@ -0,0 +1,191 @@
+package com.facebook.react;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.view.KeyEvent;
+import android.widget.Toast;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.infer.annotation.Assertions;
+import com.facebook.react.common.ReactConstants;
+import com.facebook.react.devsupport.DoubleTapReloadRecognizer;
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
+
+import javax.annotation.Nullable;
+
+/**
+ * A delegate for handling React Application support. This delegate is unaware whether it is used in
+ * an {@link Activity} or a {@link android.app.Fragment}.
+ */
+public class ReactDelegate {
+
+  public final int REQUEST_OVERLAY_PERMISSION_CODE = 1111;
+  private static final String REDBOX_PERMISSION_MESSAGE =
+    "Overlay permissions need to be granted in order for react native apps to run in dev mode.";
+  private static final String REDBOX_PERMISSION_GRANTED_MESSAGE =
+    "Overlay permissions have been granted.";
+
+
+  private final Activity mActivity;
+  private ReactRootView mReactRootView;
+
+
+  @Nullable
+  private final String mMainComponentName;
+
+  @Nullable
+  private Bundle mLaunchOptions;
+
+  @Nullable
+  private DoubleTapReloadRecognizer mDoubleTapReloadRecognizer;
+
+
+  public ReactDelegate(Activity activity, String appKey, Bundle launchOptions) {
+    mActivity = activity;
+    mMainComponentName = appKey;
+    mLaunchOptions = launchOptions;
+    mDoubleTapReloadRecognizer = new DoubleTapReloadRecognizer();
+  }
+
+  public void onHostResume() {
+    if (getReactNativeHost().hasInstance()) {
+      if (mActivity instanceof DefaultHardwareBackBtnHandler) {
```

Comment:
That would be my first choice. I wonder if the the behaviour would be the same, since `ReactActivity` is calling `super.onBackPressed()` and the fragment only calls `getActivity().onBackPressed()`. The chain of thought is that the host activity will have to call the fragment's `onBackPressed` method do decide whether or not to call the super method.